writer_test.go

```diff
@@ -81,3 +81,28 @@ func TestOpen(t *testing.T) {
 		assert.Equal(t, tt.filenames, names, "Opened unexpected files given paths %v.", tt.paths)
 	}
 }
+
+type testWriter struct {
+	expected []byte
```

Comment:
If we make this a string, we'll get better output on test failures and easier initialization.